### ISSUE TITLE
"Error retrieving template 'Template is not available for api version == 2022-06-30-preview'. Please try again later." when running validate-messages

### ISSUE THREAD

**Issue by scout208** [Azure/azure-iot-cli-extension#719]
When running the validate-messages command, I'm getting the error "Error retrieving template 'Template is not available for api version == 2022-06-30-preview'. Please try again later.".

![Image](https://github.com/user-attachments/assets/34e2a646-5ee2-4708-a597-75c4ccd17cec)

**Comment by yang-hai-feng** [comment]
@scout208 would you please share your application id, so I can take a look?

**Comment by scout208** [comment]
@yang-hai-feng, sure. The application id is `ef0c7748-eaf7-4749-b24f-b810882ebbd6`

**Comment by yang-hai-feng** [comment]
@scout208 do you have a device template associated with your device?  from the error information, it seems your don't have template associated with the device.

**Comment by scout208** [comment]
@yang-hai-feng Yes, I do have a device template associated with my device. I'm not sure why I'm getting this error. Running the command with the `--debug` flag show the actual API endpoint being used and when I run it in Postman it is able to get the device template.

![Image](https://github.com/user-attachments/assets/f14ffa62-6044-43b7-b87a-e9be0a9b2eeb)

![Image](https://github.com/user-attachments/assets/fb0b8c56-a86a-4304-8b82-3438de223aa0)

**Comment by yang-hai-feng** [comment]
Thanks @scout208  for the details, looks like a bug from CLI extension code,  I will investigate and get back to you.

**Comment by scout208** [comment]
@yang-hai-feng I've narrowed the problem down to this block of code 

https://github.com/Azure/azure-iot-cli-extension/blob/a42f57ae26b9ec7fc46446fae778c6de35cb94e2/azext_iot/central/models/v2022_06_30_preview/template.py#L94-L103

In my template there are no components, which means that `component_schema_names` is never getting assigned so it is throwing an error when trying to append it to the modules list.

I fixed it on my machine by changing the block to this
````
component_schema_names = []
if components:
    component_schema_names = self._extract_schema_names(components)


modules.append({
    'id': id,
    'interfaces': interfaces,
    'schema_names': schema_names,
    'components': components,
    'component_schema_names': component_schema_names
})
````

and it was able to get my device template successfully.

**Comment by yang-hai-feng** [comment]
@scout208 , thanks for your investigation, that was fantastic, we will include the fixes for next releases,  appreciated for your help to identify this issue.
